README.rst

```diff
@@ -593,6 +593,216 @@ For example, if we want to change the verbose name to "Hotspot", we could write:
+<https://github.com/openwisp/openwisp-users#authenticating-with-the-user-token>`_.
+
+When browsing the API via the `Live documentation <#live-documentation>`_
+or the `Browsable web page <#browsable-web-interface>`_, you can also use
+the session authentication by logging in the django admin.
+
+Pagination
+~~~~~~~~~~
+
+All *list* endpoints support the ``page_size`` parameter that allows paginating
+the results in conjunction with the ``page`` parameter.
+
+.. code-block:: text
+
+    GET /api/v1/controller/template/?page_size=10
+    GET /api/v1/controller/template/?page_size=10&page=2
+
+List of endpoints
+~~~~~~~~~~~~~~~~~
+
+Since the detailed explanation is contained in the `Live documentation <#live-documentation>`_
+and in the `Browsable web page <#browsable-web-interface>`_ of each point,
+here we'll provide just a list of the available endpoints,
+for further information please open the URL of the endpoint in your browser.
+
+List templates
+^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/template/
+
+Create template
+^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    POST /api/v1/controller/template/
+
+Get template detail
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/template/{pk}/
+
+Download template configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/template/{pk}/configuration/
+
+Change details of template
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    PUT /api/v1/controller/template/{pk}/
+
+Patch details of template
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    PATCH /api/v1/controller/template/{pk}/
+
+Delete template
+^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    DELETE /api/v1/controller/template/{pk}/
+
+List vpns
+^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/vpn/
+
+Create vpn
+^^^^^^^^^^
+
+.. code-block:: text
+
+    POST /api/v1/controller/vpn/
+
+Get vpn detail
+^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/vpn/{pk}/
+
+Download vpn configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/vpn/{pk}/configuration/
+
+Change details of vpn
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    PUT /api/v1/controller/vpn/{pk}/
+
+Patch details of vpn
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    PATCH /api/v1/controller/vpn/{pk}/
+
+Delete vpn
+^^^^^^^^^^
+
+.. code-block:: text
+
+    DELETE /api/v1/controller/vpn/{pk}/
+
+List devices
+^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/device/
+
+Create device
+^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    POST /api/v1/controller/device/
+
+Get device detail
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/device/{pk}/
+
+Download device configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    GET /api/v1/controller/device/{pk}/configuration/
+
+Change details of device
+^^^^^^^^^^^^^^^^^^^^^^^^
+
```

Comment:
we have to briefly explain how to assign, unassign and/or reorder templates here

Reply:
Done